### PR TITLE
Consul secret engine

### PR DESCRIPTION
**Pull request by Dudesons**
Add support for [consul secret engine](https://www.vaultproject.io/api/secret/consul/index.html)

### CODE REVIEW

**Comment by Dudesons** [comment]
Hi @TerryHowe,

I don't understand why the CI fail, in my local environment the idempotency works well.
```
ansible-playbook test_consul_role.yml
[DEPRECATION WARNING]: hashivault_secret_enable is kept for backwards compatibility but usage is discouraged. The module documentation details page may explain more about this rationale.. This feature will be removed in a future release. Deprecation warnings can 
be disabled by setting deprecation_warnings=False in ansible.cfg.

PLAY [localhost] *******************************************************************************************************************************************************************************************************************************************************

TASK [enable consul secret engine] *************************************************************************************************************************************************************************************************************************************
ok: [localhost]

TASK [successfully configure] ******************************************************************************************************************************************************************************************************************************************
changed: [localhost]

TASK [fail write a role] ***********************************************************************************************************************************************************************************************************************************************
ok: [localhost]

TASK [assert] **********************************************************************************************************************************************************************************************************************************************************
ok: [localhost] => {
    "changed": false, 
    "msg": "All assertions passed"
}

TASK [successfully write a role] ***************************************************************************************************************************************************************************************************************************************
changed: [localhost]

TASK [assert] **********************************************************************************************************************************************************************************************************************************************************
ok: [localhost] => {
    "changed": false, 
    "msg": "All assertions passed"
}

TASK [skip write a role] ***********************************************************************************************************************************************************************************************************************************************
ok: [localhost]

TASK [assert] **********************************************************************************************************************************************************************************************************************************************************
ok: [localhost] => {
    "changed": false, 
    "msg": "All assertions passed"
}

TASK [delete role] *****************************************************************************************************************************************************************************************************************************************************
changed: [localhost]

TASK [assert] **********************************************************************************************************************************************************************************************************************************************************
ok: [localhost] => {
    "changed": false, 
    "msg": "All assertions passed"
}

TASK [skip delete role] ************************************************************************************************************************************************************************************************************************************************
ok: [localhost]

TASK [assert] **********************************************************************************************************************************************************************************************************************************************************
ok: [localhost] => {
    "changed": false, 
    "msg": "All assertions passed"
}

PLAY RECAP *************************************************************************************************************************************************************************************************************************************************************
localhost                  : ok=12   changed=3    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
```

**Comment by drewmullen** [comment]
to further see what travis is doing, set the play to `-vvv` in `run.sh`. then we can verify the values

**Comment by Dudesons** [comment]
@drewmullen I would like to implement the idempotency but I didn't find a request to fetch the config :( consul [secret engine doc](https://www.vaultproject.io/api/secret/consul/index.html)
Maybe you have an idea how to do that

**Comment by drewmullen** [comment]
can you hit consul/config with a GET? i havent used that secret engine

**Comment by drewmullen** [comment]
actually, i think what youre looking for is: https://www.vaultproject.io/api/system/mounts.html#read-mount-configuration

`sys/mounts/consul/tune`

File: functional/test_consul_config.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        address: "{{ address }}"
+        scheme: "{{ scheme }}"
+        consul_token: "{{ consul_token }}"
+      register: fail_config
+      failed_when: false
+
+    - assert: { that: "{{ fail_config.changed }} == False" }
+    
+    - name: enable database secret engine
+      hashivault_secret_enable:
```
Comment (drewmullen) [inline functional/test_consul_config.yml]: this module is deprecated now, please use `hashivault_secret_engine`

Comment (drewmullen) [reply functional/test_consul_config.yml]: fyi - fixing for future ppl too https://github.com/TerryHowe/ansible-modules-hashivault/pull/159/commits/c4be6a82735296f2bd13012c1cc973609003e614

**Comment by Dudesons** [comment]
@drewmullen I did some tests I think the `hashivault_consul_secret_engine_config ` can't be idempotent :(
I try with this procedure to check that:
1) request  `/sys/mounts/my-mount/tune` to know if the mount point exists
2) request GET on consul/config with that I can check the address and the scheme

After that I can't check the consul token provided for the setup, this is the unique parameter I can't check

**Comment by drewmullen** [comment]
```
curl -H 'X-Vault-Token: root'  http://127.0.0.1:8200/v1/consul/config/access
{"request_id":"658beacc-0f9d-d3ae-384a-aa3cfde2d45b","lease_id":"","renewable":false,"lease_duration":0,"data":{"address":"127.0.0.1:8500","scheme":"http"},"wrap_info":null,"warnings":null,"auth":null}
```

**Comment by TerryHowe** [comment]
Where are we on this, just missing idempotence?

**Comment by Dudesons** [comment]
@TerryHowe we can implement idempotence on `hashivault_consul_secret_engine_config`but the day you want to change the consul token of this configuration it won't work because we can't check if the token has changed for security reason.
So here we have 2 choices:
 * Don't implement idempotence but the action can really update the configuration
 * Implement idempotence but the configuration is managed partially (if you update the token you are stucked)

**Comment by TerryHowe** [comment]
I'll fix the problems here, this has sat around long enough. Thanks!

**Comment by TerryHowe** [comment]
4.4.7

**Pull request merged**